client/plots/matrix.layout.js

```diff
@@ -233,6 +241,24 @@ export function setLabelsAndScales() {
 			t.label = `${t.label} (${count})`
 		}
 		if (t.grp.type !== 'hierCluster' && t.tw.q?.mode == 'continuous') {
+			const vc = t.tw.term.valueConversion
+			if (vc) {
+				// convert values
+				t.counts.minval *= vc.scaleFactor
+				t.counts.maxval *= vc.scaleFactor
+			}
+
+			if (renderedContinuousVs.length && t.tw.q.convert2ZScore) {
```

Comment:
how about you look into carrying out zscore computing in getData() so that no need to do all these on client. also that will make zscore to work for violin

Reply:
mean and std were calculated using the renderedValues, which are decided on client side (classifyValues). So cannot calculate zscore in getData() since at that time we don't know which values will be rendered yet.

Reply:
@xzhou82 my opinion: I think it's better to compute on the client side, instead of asking the server to transform the values when the data source does not actually change. That means re-reading db/files/external API for the same data on the server, it would impact the server less (by allowing caching and offloading computation) if the value transformation is done on the client.  Also, in case some of the data/rendered filters are done on the client-side plot, then the server may not know which data are actually rendered, meaning the mean and stdev may not be computable ton the server.

Reply:
ok to merge. since the new boolean flag is added to tw.q{}, it becomes part of request payload and will incur more requests since payload are now different

Reply:
Congyu should add these notes as comments otherwise i will be wondering about it again